### ISSUE TITLE
Mixed precision does not work

### ISSUE THREAD

**Issue by wangxiaoyuwangdayu** [ibrahimethemhamamci/CT-CLIP#18]
I printed the tensor dtype when training,  and they were all float32. It seems that mixed precision does not work.

**Comment by sezginerr** [comment]
Yes, mixed precision does not work with the code. I also have some experiences manually applying it as well, but training diverges. So I would recommend not to use it.
